api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ValueSetTranslatorImpl.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import javax.annotation.Nonnull;
+
+import java.util.Optional;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.Enumerations;
+import org.hl7.fhir.r4.model.ValueSet;
+import org.openmrs.Concept;
+import org.openmrs.ConceptMap;
+import org.openmrs.ConceptReferenceTerm;
+import org.openmrs.ConceptSet;
+import org.openmrs.module.fhir2.api.FhirConceptSourceService;
+import org.openmrs.module.fhir2.api.translators.ValueSetReferenceTranslator;
+import org.openmrs.module.fhir2.api.translators.ValueSetTranslator;
+import org.openmrs.module.fhir2.model.FhirConceptSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class ValueSetTranslatorImpl implements ValueSetTranslator {
+	
+	@Autowired
+	private FhirConceptSourceService conceptSourceService;
+	
+	@Autowired
+	private ValueSetReferenceTranslator valueSetReferenceTranslator;
+	
+	@Override
+	public ValueSet toFhirResource(@Nonnull Concept concept) {
+		if (concept == null) {
+			return null;
+		}
+		
+		if (!concept.getSet()) {
+			return null;
+		}
+		
+		ValueSet valueSet = new ValueSet();
+		valueSet.setId(concept.getUuid());
+		valueSet.setTitle(Optional.ofNullable(concept.getName()).map(n -> n.getName()).orElse(""));
+		valueSet.setStatus(Enumerations.PublicationStatus.ACTIVE);
+		
+		valueSet.setDate((concept.getDateChanged() != null) ? concept.getDateChanged() : concept.getDateCreated());
+		
+		String description = (concept.getDescription() == null || concept.getDescription().getDescription() == null) ? ""
+		        : concept.getDescription().getDescription();
+		valueSet.setDescription(description);
+		
+		ValueSet.ValueSetComposeComponent compose = new ValueSet.ValueSetComposeComponent();
+		
+		for (ConceptSet conceptSet : concept.getConceptSets()) {
+			Concept conceptSetMember = conceptSet.getConcept();
+			
+			if (conceptSetMember != null && conceptSetMember.getSet()) {
+				ValueSet.ConceptSetComponent conceptSetComponent = new ValueSet.ConceptSetComponent();
+				conceptSetComponent.addValueSet(valueSetReferenceTranslator.toFhirResource(conceptSetMember).getReference());
```

Comment:
@ibacher is it correct to add reference like this or it should be an absolute URL?

Reply:
@Richa-09 Normally, it would be fine to use a relative reference. However, in this case because the FHIR spec says that the value is "canonical" rather than a "reference", I do think we need to use the full URL.

Reply:
@ibacher Since base URL will vary for different implementers, is there any way to get the base URL? or should I hard code?

Reply:
@Richa-09 Ooph... good point. Let's leave it as a standard reference for now. There is a way to this, but it will be messy.

The "right way" would be to move the [OpenmrsFhirAddressStrategy](https://github.com/openmrs/openmrs-module-fhir2/blob/master/omod/src/main/java/org/openmrs/module/fhir2/web/util/OpenmrsFhirAddressStrategy.java) class into the API module, capture the raw `HttpServletRequest` as a parameter in the `ResourceProvider` and then call [`determineServerBase`](https://github.com/openmrs/openmrs-module-fhir2/blob/290688fc9189b0fa108c77f09f46c2560292e84d/omod/src/main/java/org/openmrs/module/fhir2/web/util/OpenmrsFhirAddressStrategy.java#L83) with the request. Let me think if there's a better way to implement things.